cvat/apps/webhooks/signals.py

```diff
@@ -0,0 +1,204 @@
+        return instance.id
+
+    try:
+        pid = getattr(instance, 'project_id', None)
+        if pid is None:
+            return instance.get_project_id()
+        return pid
+    except Exception:
+        return None
+
+def organization_id(instance):
+    if isinstance(instance, Organization):
+        return instance.id
+
+    try:
+        oid = getattr(instance, 'organization_id', None)
+        if oid is None:
+            return instance.get_organization_id()
+        return oid
+    except Exception:
+        return None
+
+@receiver(signal_update)
+def update(sender, instance=None, old_values=None, **kwargs):
+    event = event_name("update", sender.basename)
+    if event not in map(lambda a: a[0], EventTypeChoice.choices()):
+        return
+
+    serializer = sender.get_serializer_class()(
+        instance=instance,
+        context={"request": sender.request}
+    )
+
+    pid = project_id(instance)
+    oid = organization_id(instance)
+
+    if not any((oid, pid)):
+        return
+
+    data = {
+        "event": event,
+        sender.basename: serializer.data,
+        "before_update": old_values,
+    }
+
+    for webhook in select_webhooks(pid, oid, event):
+        data.update({"webhook_id": webhook.id})
+        add_to_queue(webhook, payload(data, sender.request))
+
+@receiver(signal_create)
+def resource_created(sender, instance=None, **kwargs):
+    event = event_name("create", sender.basename)
+    if event not in map(lambda a: a[0], EventTypeChoice.choices()):
+        return
+
+    pid = project_id(instance)
+    oid = organization_id(instance)
+    if not any((oid, pid)):
+        return
+
+    serializer = sender.get_serializer_class()(
+        instance=instance,
+        context={"request": sender.request}
+    )
+
+    data = {
+        "event": event,
+        sender.basename: serializer.data
+    }
+
+    for webhook in select_webhooks(pid, oid, event):
+        data.update({"webhook_id": webhook.id})
+        add_to_queue(webhook, payload(data, sender.request))
+
+
+@receiver(signal_delete)
+def resource_deleted(sender, instance=None, **kwargs):
+    event = event_name("delete", sender.basename)
+    if event not in map(lambda a: a[0], EventTypeChoice.choices()):
+        return
+
+    pid = project_id(instance)
+    oid = organization_id(instance)
+    if not any((oid, pid)):
+        return
+
+    serializer = sender.get_serializer_class()(
+        instance=instance,
+        context={"request": sender.request}
+    )
+
+    data = {
+        "event": event,
+        sender.basename: serializer.data
+    }
+
+    for webhook in select_webhooks(pid, oid, event):
+        data.update({"webhook_id": webhook.id})
+        add_to_queue(webhook, payload(data, sender.request))
+
+
+@receiver(signal_redelivery)
+def redelivery(sender, data=None, **kwargs):
+    add_to_queue(sender.get_object(), data, redelivery=True)
+
+
+@receiver(signal_ping)
+def ping(sender, serializer=None, **kwargs):
```

Comment:
I don't think serializer can be None here.

Reply:
Fixed